lighthouse-core/computed/metrics/navigation-metric.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license Copyright 2018 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Metric = require('./metric.js');
+
+/**
+ * @fileOverview Enforces that a metric can only be computed on navigations.
+ */
+class NavigationMetric extends Metric {
+  /**
+   * @param {LH.Artifacts.NavigationMetricComputationData} data
+   * @param {LH.Artifacts.ComputedContext} context
+   * @return {Promise<LH.Artifacts.LanternMetric>}
+   */
+  static computeSimulatedMetric(data, context) { // eslint-disable-line no-unused-vars
```

Comment:
Can we get rid of `computeSimulatedMetric` and `computeObservedMetric` now that we are extending `Metric`?

Reply:
We can, but I thought it was useful to still capture the narrower type of `data`.  All of this inheritance is really in "square is not a rectangle" territory anyway which is why I opted for the wonky `Metric._compute.call` style.

I'm up for either.

Reply:
> I thought it was useful to still capture the narrower type of `data`

This is a good enough reason to keep it, although I wish we could enforce this type when the functions are redefined in a child class.

Leaving as is SGTM